src/highdicom/seg/sop.py

```diff
@@ -944,3 +966,1458 @@ def _encode_pixels(self, planes: np.ndarray) -> bytes:
+                desc.segmented_property_type == segmented_property_type
+            )
+        if algorithm_type is not None:
+            algo_type = SegmentAlgorithmTypeValues(algorithm_type)
+            filter_funcs.append(
+                lambda desc:
+                SegmentAlgorithmTypeValues(desc.algorithm_type) == algo_type
+            )
+        if tracking_uid is not None:
+            filter_funcs.append(
+                lambda desc: desc.tracking_uid == tracking_uid
+            )
+        if tracking_id is not None:
+            filter_funcs.append(
+                lambda desc: desc.tracking_id == tracking_id
+            )
+
+        matches = [
+            desc.segment_number
+            for desc in self.SegmentSequence
+            if all(f(desc) for f in filter_funcs)
+        ]
+
+        return matches
+
+    def search_for_tracking_ids(
+        self,
+        segmented_property_category: Optional[Union[Code, CodedConcept]] = None,
+        segmented_property_type: Optional[Union[Code, CodedConcept]] = None,
+        algorithm_type: Optional[Union[SegmentAlgorithmTypeValues, str]] = None
+    ) -> List[str]:
+        """Get all unique tracking identifiers in this SEG image.
+
+        Any number of optional filters may be provided. A segment must match
+        all provided filters to be included in the returned list.
+
+        Parameters
+        ----------
+        segmented_property_category: Optional[Union[Code, CodedConcept]], optional
+            Segmented property category filter to apply.
+        segmented_property_type: Optional[Union[Code, CodedConcept]], optional
+            Segmented property type filter to apply.
+        algorithm_type: Optional[Union[SegmentAlgorithmTypeValues, str]], optional
+            Segmented property type filter to apply.
+
+        Returns
+        -------
+        List[str]
+            All unique tracking identifiers referenced in segment descriptions
+            in this SEG image that match all provided filters.
+
+        """  # noqa: E501
+        filter_funcs = []
+        if segmented_property_category is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_category == segmented_property_category
+            )
+        if segmented_property_type is not None:
+            filter_funcs.append(
+                lambda desc:
+                desc.segmented_property_type == segmented_property_type
+            )
+        if algorithm_type is not None:
+            algo_type = SegmentAlgorithmTypeValues(algorithm_type)
+            filter_funcs.append(
+                lambda desc:
+                SegmentAlgorithmTypeValues(desc.algorithm_type) == algo_type
+            )
+
+        return list({
+            desc.tracking_id for desc in self.SegmentSequence
+            if desc.tracking_id is not None and
+            all(f(desc) for f in filter_funcs)
+        })
+
+    def search_for_tracking_uids(
```

Comment:
```suggestion
    def get_tracking_uids(
```

Reply:
Resolved in 9f466a42080777e8943d4e00f5f2593a1977adaf

Reply:
Resolved in df7cb4801c8e090870e30e90d29e992e0e62efb3